### PR TITLE
Add support for SKU Database to support updates

### LINKED ISSUE #1229: Bug: Updating SKU/Edition field of AzureSQLDatabase doesn't do anything
**Describe the bug**
When updating the SKU/Edition field of the AzureSQLDatabase CRD, the updated field isn't getting propagated to Azure but the CRD switches to Provisioned status which is misleading.

**To Reproduce**
1. Deploy AzureSQLDatabase
2. Update SKU/Edition field, observe that it doesn't get applied to the backing Azure resource

**Expected behavior**
The update is applied

### PR DESCRIPTION
**Pull request by matthchr**
Closes #1229

**What this PR does / why we need it**:
Adds support for updating SQL Databases

**Special notes for your reviewer**:
Also improved test resource cleanup and usage of the ARM LRO polling client a bit

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,6 +97,11 @@ test-cleanup-azure-resources:
 		az group delete --name $$rgname --no-wait --yes; \
 	done
 
+	for rgname in `az group list --query "[*].[name]" -o table | grep 'rg-prime$$' `; do \
+    	echo "$$rgname will be deleted"; \
+    	az group delete --name $$rgname --no-wait --yes; \
+    done
```
Comment (frodopwns) [inline Makefile:103]: need an explanation here since this change appears to be out of scope

Comment (matthchr) [reply Makefile:103]: I can break it out into it's own PR.

Basically, running the tests can leak resources (at least locally?) because the tests are suffixed with rg-prime. Let me know if you want me to move this to its own PR.

File: Makefile [hunk 2]
```diff
@@ -97,6 +97,11 @@ test-cleanup-azure-resources:
 		az group delete --name $$rgname --no-wait --yes; \
 	done
 
+	for rgname in `az group list --query "[*].[name]" -o table | grep 'rg-prime$$' `; do \
```
Comment (Porges) [inline Makefile:100]: 💄  inconsistent indentation

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go [hunk 3]
```diff
@@ -18,6 +18,9 @@ import (
 type AzureSqlDbManager struct {
 }
 
+// Ensure we implement the interface we expect
+var _ SqlDbManager = &AzureSqlDbManager{}
```
Comment (frodopwns) [inline pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go:22]: I'm not familiar with this syntax, what is the "_" doing? I would say its generally more go-like to use the New function than to use a global constant.

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go [hunk 4]
```diff
@@ -69,66 +66,107 @@ func (db *AzureSqlDbManager) Ensure(ctx context.Context, obj runtime.Object, opt
...
 		// resource request has been sent to ARM
 		if azerr.Type == errhelp.AsyncOpIncompleteError {
+			instance.Status.Message = "Resource request successfully submitted to Azure"
 			instance.Status.Provisioning = true
+			instance.Status.PollingURL = pollingUrl
 			return false, nil
 		}
 
+		if azerr.Type == errhelp.InvalidMaxSizeTierCombination {
+			instance.Status.FailedProvisioning = true
```
Comment (Porges) [inline pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go:165]: should we hide these two status fields from modification and instead provide methods that update them correctly?

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go (hunk 3 above)
Comment (matthchr) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go:22]: This is a type assertion asking the compiler to ensure that we've implemented this interface. The compiler elides it at compile time (because we're assigning to a `_` variable).

It's useful in places where you have a type that impl's an interface and want a more C# (or pick your OO language of choice) interface (i.e. the type implementing the interface knows that it is doing so).

In this case, one of my previous PRs had actually changed a method signature on the AzureSqlDbManager struct but missed changing it in the SqlDbManager interface -- this is just a trick so that the compiler catches that if we make the same mistake in the future.

Comment (Porges) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go:22]: it is asserting that the interface `SqlDbManager` is implemented correctly; if it isn’t then this line will fail to compile

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go (hunk 4 above)
Comment (matthchr) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go:165]: We probably should, yes -- and maybe in the future consider merging some of them (i.e. `Provisioning` and `FailedProvisioning` could be like `ProvisioningStatus` as an enum instead of 2 bools that represent 3 distinct states (since 1 of the 4 is invalid).

Comment (matthchr) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go:165]: Probably in a different PR though>

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go (hunk 3 above)
Comment (Porges) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb.go:22]: [snap] 🐊

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go (hunk 4 above)
Comment (Porges) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go:165]: sure

Comment (matthchr) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go:165]: I'll file an issue for it

**Comment by frodopwns** [comment]
Created a Basic DB and then attempted to update it with a new MaxSize. Got this error: `Code="InvalidMaxSizeTierCombination" Message="The tier 'Basic' does not`

Bit, failedProvisioning was not set and the operator continued to reconcile continuously.

```
status:
  message: Code="InvalidMaxSizeTierCombination" Message="The tier 'Basic' does not
    support the database max size '4294967296'."
  pollingUrl: https://management.azure.com/subscriptions/7/resourceGroups/resourcegroup-skukazoo/providers/Microsoft.Sql/locations/westus/databaseAzureAsyncOperation/d0d85658-2f37-4426-8312-f2f15362f019?api-version=2017-10-01-preview
  resourceId: /subscriptions/e9facfac/resourceGroups/resourcegroup-skukazoo/providers/Microsoft.Sql/servers/sqlserver-skukazoo/databases/azuresqldatabase-skukazoo
  specHash: f48f4f25ea852ea25a850285d7317ad3cda293efb19017d430551cce552f785c
  state: Online
```

File: pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go (hunk 4 above)
Comment (matthchr) [reply pkg/resourcemanager/azuresql/azuresqldb/azuresqldb_reconcile.go:165]: https://github.com/Azure/azure-service-operator/issues/1243

**Comment by matthchr** [comment]
I fixed the issue @frodopwns mentioned where the state wasn't set quite right.
